drivers: firmware: nrf_ironside: Minor fixes

57300 opened the pull request:
* Guard behind `CONFIG_NRF_IRONSIDE`
* Make events static

57300 committed:
drivers: firmware: nrf_ironside: Make events static

Events used in `call.c` should've been defined as static.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

pdgendt commented on drivers/firmware/nrf_ironside/CMakeLists.txt:
```diff
@@ -2,5 +2,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
+zephyr_library_property(ALLOW_EMPTY TRUE)
```
Can't we make the inclusion conditional somehow? Instead of adding an empty library to each build.

pdgendt replied:
I think a (hidden) symbol should be introduced

```
config NRF_IRONSIDE
	bool
```

And then change `drivers/CMakeLists.txt` with a conditional include

```diff
diff --git a/drivers/firmware/CMakeLists.txt b/drivers/firmware/CMakeLists.txt
index 167d21addb5..8380b3f08c5 100644
--- a/drivers/firmware/CMakeLists.txt
+++ b/drivers/firmware/CMakeLists.txt
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0

 # zephyr-keep-sorted-start
-add_subdirectory(nrf_ironside)
+add_subdirectory_ifdef(CONFIG_NRF_IRONSIDE nrf_ironside)
 add_subdirectory_ifdef(CONFIG_ARM_SCMI scmi)
 # zephyr-keep-sorted-stop
```

57300 committed:
drivers: firmware: nrf_ironside: Guard behind CONFIG_NRF_IRONSIDE

Add a promptless symbol that can be selected by different drivers. It is
there to ensure that the corresponding library is not added to the build
unless it contains at least one source file.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>
